### PR TITLE
45216 continuousbenchmarking nuketmp 20161006 5

### PR DESCRIPTION
**Pull request by ashawnbandy-te-tfb**
As it stands currently, many frameworks are writing to /tmp. So many, in fact, that in our testing environment the pool of inodes are exhausted which causes tests needing to write anything to fail. This commit introduces work to remove files that are generated by a test after that test is done. Note: there is no attempt to remove all files from /tmp (i.e. sudo is not used, nor is the search recursive).

### CODE REVIEW

**Comment by knewmanTE** [comment]
Could this cause any problems for users who aren't running the testing suite in an isolated VM?

**Comment by knewmanTE** [comment]
@ashawnbandy-te-tfb document telling users that they should be running in an isolated environment, or telling users that it will only be deleting tmp files owned by testrunner?

**Comment by ashawnbandy-te-tfb** [comment]
I would say probably a caution that one should not use one's regular user to run tests in an environment where erasing files out of /tmp would be a problem.  The worst case should not be all that bad: /tmp is meant to be temporary in nature.

File: toolset/benchmark/benchmarker.py [hunk 1]
```diff
@@ -650,6 +650,19 @@ def exit_with_code(code):
         time.sleep(5)
 
         ##########################################################
+        # Remove contents of  /tmp folder
+        ##########################################################
+        try:
+          filelist = [ f for f in os.listdir("/tmp") ]
+	  for f in filelist:
```
Comment (LadyMozzarella) [inline toolset/benchmark/benchmarker.py]: I just wanted to let you know that there are a couple of tab characters mixed in here. There's one at the beginning of the line here and two lines down (`os.remove("/tmp/ + f")`).

Comment (ashawnbandy-te-tfb) [inline toolset/benchmark/benchmarker.py]: Good catch!  Will correct.

**Comment by ashawnbandy-te-tfb** [comment]
@knewmanTE suggested adding a command line switch to turn on this feature (i.e. by default no files are removed).  I will add that and re-open this PR when it is complete.

**Comment by knewmanTE** [comment]
@ashawnbandy-te-tfb you can also just add "WIP" to the title of the task to indicate that it is a work in progress!

**Pull request merged**